### PR TITLE
[FIX]point_of_sale: BlackBox BE

### PR DESCRIPTION
**Pull request by rhe-odoo**
Update the blackbox from v11 in v13.
We had to make a new driver to be able to use the blackbox with the iotbox.
We also had to make the new features compatible with the blackbox. 

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/pos_hr/static/src/js/chrome.js [hunk 1]
```diff
@@ -95,7 +95,7 @@ odoo.define('pos_hr.chrome', function (require) {
             'args': {
                 label: _lt('Lock'),
                 action: function() {
-                    this.gui.show_screen('login');
+                    this.chrome.return_to_login_screen();
```
Comment (pimodoo) [inline addons/pos_hr/static/src/js/chrome.js]: why changin this?

File: addons/hw_drivers/drivers/SerialBaseDriver.py [hunk 2]
```diff
@@ -91,7 +91,7 @@ def _set_name(self):
         """Tries to build the device's name based on its type and protocol name but falls back on a default name if that doesn't work."""
 
         try:
-            name = ('%s serial %s' % (self._protocol.name, self._device_type)).title()
+            name = self._protocol.name.title()
```
Comment (pimodoo) [inline addons/hw_drivers/drivers/SerialBaseDriver.py]: @aprieels can you check this commit?

Comment (aprieels) [reply addons/hw_drivers/drivers/SerialBaseDriver.py]: Yes, this change is not really useful and will change the name of connected Serial devices in stable so I don't think we need it

File: addons/pos_hr/static/src/js/chrome.js (hunk 1 above)
Comment (rhe-odoo) [reply addons/pos_hr/static/src/js/chrome.js]: The function will be override when using employee in blackbox. Checking if the user clocked out before showing the login screen.

File: addons/point_of_sale/views/pos_config_view.xml [hunk 3]
```diff
@@ -262,7 +262,7 @@
                                 </div>
                             </div>
                         </div>
-                        <div class="col-12 col-lg-6 o_setting_box" id="pricelist_setting" groups="product.group_product_pricelist">
+                        <div class="col-12 col-lg-6 o_setting_box" name="pricelist_setting" groups="product.group_product_pricelist">
```
Comment (pimodoo) [inline addons/point_of_sale/views/pos_config_view.xml]: you cannot change this in stable and you don't use it so remove

File: addons/point_of_sale/views/pos_config_view.xml [hunk 4]
```diff
@@ -293,7 +293,7 @@
                                 </div>
                             </div>
                         </div>
-                        <div class="col-xs-12 col-lg-6 o_setting_box" >
+                        <div class="col-xs-12 col-lg-6 o_setting_box" name="pos_discount_settings">
```
Comment (pimodoo) [inline addons/point_of_sale/views/pos_config_view.xml]: you cannot change this in stable and you don't use it so remove

File: addons/point_of_sale/views/pos_config_view.xml [hunk 5]
```diff
@@ -309,7 +309,7 @@
                                 </div>
                             </div>
                         </div>
-                        <div class="col-12 col-lg-6 o_setting_box">
+                        <div class="col-12 col-lg-6 o_setting_box" name="pos_loyalty_settings">
```
Comment (pimodoo) [inline addons/point_of_sale/views/pos_config_view.xml]: you cannot change this in stable and you don't use it so remove

File: addons/hw_posbox_homepage/views/homepage.html [hunk 6]
```diff
@@ -125,7 +125,7 @@ <h2 class="text-center text-green">Your IoT Box is up and running</h2>
                     {% endif %}
                     {% for iot_devices in iot_device_status|groupby('type') %}
                         <div class="collapsible">
-                            <div class="title arrow-down">{{ iot_devices.grouper|capitalize }}s</div>
+                            <div class="title arrow-down">{{ iot_devices.grouper|replace("_", " ")|title }}s</div>
```
Comment (pimodoo) [inline addons/hw_posbox_homepage/views/homepage.html]: Not sure that this change is needed either

File: addons/point_of_sale/models/pos_session.py [hunk 7]
```diff
@@ -542,7 +542,7 @@ def get_income_account(order_line):
         price = order_line.price_unit * (1 - (order_line.discount or 0.0) / 100.0)
         taxes = tax_ids.compute_all(price_unit=price, quantity=order_line.qty, currency=self.currency_id, is_refund=order_line.qty<0).get('taxes', [])
         date_order = order_line.order_id.date_order
-        taxes = [{'date_order': date_order, **tax} for tax in taxes]
+        taxes = [{'date_order': date_order, **tax} for tax in taxes if abs(tax['amount']) > 0]
```
Comment (pimodoo) [inline addons/point_of_sale/models/pos_session.py]: you can delete this commit, a fix had been done for the tax at 0 % in https://github.com/odoo/odoo/commit/c7936d2884c10865bfc39a43d30aef6926f29079#diff-f0aa3e0f5e29080288e68f327654dc6c

File: addons/hw_posbox_homepage/views/homepage.html (hunk 6 above)
Comment (pimodoo) [reply addons/hw_posbox_homepage/views/homepage.html]: ??

File: addons/pos_restaurant/static/src/js/splitbill.js [hunk 8]
```diff
@@ -83,60 +91,62 @@ var SplitbillScreenWidget = screens.ScreenWidget.extend({
...
-        for(var i = 0; i < orderlines.length; i++){
-            var id = orderlines[i].id;
-            var split = splitlines[id];
-            if(!split){
-                full = false;
-            }else{
-                if(split.quantity){
-                    empty = false;
-                    if(split.quantity !== orderlines[i].get_quantity()){
+    check_full_pay_order:function (order, splitlines) {
```
Comment (aprieels) [inline addons/pos_restaurant/static/src/js/splitbill.js]: Could we simplify this function to something like: 
```
check_full_pay_order: function (order, splitlines) {
    order.get_orderlines().forEach(function(orderLine) {
        var split = splitlines[orderLine.id];
        if (!split || split && split.quantity !== orderline.get_quantity()) {
            return False;
        }
    }
    return True
}
```

File: addons/pos_restaurant/static/src/js/splitbill.js [hunk 9]
```diff
@@ -83,60 +91,62 @@ var SplitbillScreenWidget = screens.ScreenWidget.extend({
...
+                order.remove_orderline(line);
+            }
+            delete splitlines[id];
         }
-        
-        if(empty){
+    },
+
+    pay: function(order,neworder,splitlines){
+        if(Object.keys(splitlines).length === 0)    // Splitlines is empty
```
Comment (aprieels) [inline addons/pos_restaurant/static/src/js/splitbill.js]: `_.isEmpty(splitlines);` ?

File: addons/point_of_sale/views/pos_config_view.xml [hunk 10]
```diff
@@ -293,7 +293,7 @@
                                 </div>
                             </div>
                         </div>
-                        <div class="col-xs-12 col-lg-6 o_setting_box" >
+                        <div class="col-xs-12 col-lg-6 o_setting_box">
```
Comment (pimodoo) [inline addons/point_of_sale/views/pos_config_view.xml]: remove this useless diff

**Comment by qle-odoo** [comment]
@robodoo r+

**Comment by qle-odoo** [comment]
@robodoo r+ rebase-ff

**Pull request closed without merging**